bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/runtime/CodeCheckEnvironmentChecker.java

```diff
@@ -161,10 +164,16 @@ public String getLogFileLocation() {
      * @param buildLog Path to the compile commands file, which the analyze command uses.
      * @return The constructed analyze command.
      */
-    public String createAnalyzeCommmand(String buildLog){
-        return codeCheckerCommand + " analyze " + getConfigValue(ConfigTypes.CHECKER_LIST) +
-             " -j "+ getConfigValue(ConfigTypes.ANAL_THREADS) + " -n javarunner" +
-             " -o "+ getConfigValue(ConfigTypes.CHECKER_WORKSPACE)+"/results/ " + buildLog;
+    public String createAnalyzeCommmand(String buildLog) {
+        return "'" + Utils.esc(codeCheckerCommand) + "'" +
```

Comment:
Wouldn't be easier to handle the command line options as a list of strings until the CommandLine type construction in the `buildScriptCommandLine`

There is a [CommandLineParser](https://commons.apache.org/proper/commons-cli/javadocs/api-release/index.html) which could be useful too.

Reply:
Thanks @gyorb for the suggestion. This section was revised, and modified according to this https://commons.apache.org/proper/commons-exec/tutorial.html . The cli parser you mentioned was being used already but extended a bit.